packages/ui/src/components/form-primitives/textarea.tsx

```diff
@@ -0,0 +1,124 @@
+import { forwardRef, TextareaHTMLAttributes, useCallback, useState } from 'react'
+
+import { ControlGroup, FormCaption, Label } from '@/components'
+import { anyTypeOf, useMergeRefs } from '@/utils'
+import { cn } from '@utils/cn'
+import { cva, VariantProps } from 'class-variance-authority'
+
+const textareaVariants = cva('cn-textarea', {
+  variants: {
+    theme: {
+      default: '',
+      danger: 'cn-textarea-danger'
+    }
+  },
+  defaultVariants: {
+    theme: 'default'
+  }
+})
+
+export interface TextareaProps extends TextareaHTMLAttributes<HTMLTextAreaElement> {
+  label?: string
+  theme?: VariantProps<typeof textareaVariants>['theme']
+  caption?: string
+  error?: string
```

Comment:
We need `warning` too
```suggestion
  error?: string
  warning?: string
```

Reply:
Fixed